### DISCUSSION TITLE
Is there a way to do an open tab counter in TST for tabs in all windows?

### DISCUSSION THREAD

**Discussion by pensive-fuzz** [piroor/treestyletab#3137] [Q&A]
I want to incorporate a tab counter into TST and I'm aware of the [tab count](https://github.com/piroor/treestyletab/wiki/Code-snippets-for-custom-style-rules#total-tab-count) code snippet. However, that snippet only counts the tabs in the current window and I'd also like to display the total tab from an entire Firefox session that has multiple windows open. I'm at a loss on how to adapt the snippet to all windows.

**Comment by piroor** [comment]
https://addons.mozilla.org/firefox/addon/tab-counter-webext/ ?

**Reply by pensive-fuzz** [reply]
I'm aware of countless other extensions that provide a tab count, but I'm specifically interested in incorporating it into one place on TST's sidebar panel.

**Reply by piroor** [reply]
It is impossible without any helper extension, because there is no hint to show all tabs count in TST's sidebar page.

TST has an API to insert arbitrary contents to the "new tab" button, so you can create a new helper extension based on it.
https://github.com/piroor/treestyletab/wiki/Extra-Tab-Contents-API#extra-contents-in-the-new-tab-button

**Reply by pensive-fuzz** [reply]
Guess I'll have to look into how to write an extension then, thanks for giving me a direction to go towards.
